### PR TITLE
Add support for SJWS01LM BLE flooding sensor

### PR DESCRIPTION
**Pull request by lis0r**
## Description:

Adds support for the Xiaomi Mi Flood Detector SJWS01LM. Based on the door sensor.

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [X] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino [hunk 1]
```diff
@@ -2985,6 +3006,15 @@ void MI32GetOneSensorJson(int slot, int hidename){
         ResponseAppend_P(PSTR(",\"Btn\":%d"),p->Btn);
       }
     }
+    if (p->feature.flooding){
+      if(p->eventType.flooding || !MI32.mode.triggeredTele || MI32.option.allwaysAggregate
+#ifdef USE_HOME_ASSISTANT
```
Comment (Jason2866) [inline tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino:3011]: This change?

Comment (lis0r) [reply tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino:3011]: This logic was copied from line 3000 onwards, for the button input as used by the door sensor. If that doesn't make sense, I can simplify it?

**Comment by barbudor** [comment]
@lis0r Does this sensor works with a key ?

**Comment by lis0r** [comment]
> @lis0r Does this sensor works with a key ?

Yes, I extracted it with https://atc1441.github.io/Temp_universal_mi_activate.html .

**Comment by Jason2866** [comment]
@Staars PTL

**Comment by Staars** [comment]
Can confirm that this sensor needs a bind_key( legacy driver supports it for about 2 years).

Should be okay.

**Pull request merged**